### DISCUSSION TITLE
Routing to illegal destinations

### DISCUSSION THREAD

**Discussion by rkleineisel** [osmandapp/OsmAnd#20996] [Ideas]
When navigating to a destination which is illegal to reach with the current vehicle I get no warning. OSMAND happily routes me into a pedestrian zone with car routing or into a road that's illegal for trucks with truck routing.
The avoidance works fine as long as the destination is not inside the illegal road. In this case I get routed around it.
I think it would be very helpful to know at the start of navigation that the destination cannot be reached with the current vehicle legally.

**Comment by DmitryAlexei** [comment]
Could you please provide test cases from our map (https://osmand.net/map/)
Also please specify the version of the app you use

**Reply by rkleineisel** [reply]
My app version is 4.8.6 on Android.

The routing in the online map behaves slightly different: Here the routing line ends at the start of the pedestrian zone.

On example would be routing with the route profile "car" from:
49.59427, 11.00332
to 
49.59634, 11.00514

The destination is in the "Untere Karlstasse" in Erlangen, Germany, which is a "highway=pedestrian" (https://www.openstreetmap.org/way/4047333).

When calculating this same route in the app the routing leads right into the pedestrian zone.

**Reply by DmitryAlexei** [reply]
Yes, there is a difference OsmAnd map > iOS app
https://osmand.net/map/navigate/?start=49.594217,11.003408&end=49.596340,11.005140&profile=car#18/49.59517/11.00444
<img width="706" alt="Screenshot 2024-10-10 at 13 21 18" src="https://github.com/user-attachments/assets/d4db0bc7-0df0-43e5-8d2e-bf607b18bf3e">
You are referring to issue https://github.com/osmandapp/OsmAnd/issues/17987
It might be fixed in Android in the next release
